### PR TITLE
irods: 4.2.7 -> 4.2.11

### PR DESCRIPTION
**Pull request by bzizou**
This commit should work once #129416 and #161196 merged

###### Motivation for this change
- upgrade irods
- fix as a dependency of #129416

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/filesystems/irods/common.nix [hunk 1]
```diff
@@ -28,6 +32,7 @@
     substituteInPlace CMakeLists.txt --replace "DESTINATION usr/bin" "DESTINATION bin"
     substituteInPlace CMakeLists.txt --replace "INCLUDE_DIRS usr/include/" "INCLUDE_DIRS include/"
     substituteInPlace CMakeLists.txt --replace "DESTINATION usr/lib/" "DESTINATION lib/"
+    substituteInPlace CMakeLists.txt --replace "{IRODS_EXTERNALS_FULLPATH_JSON}/include" "{IRODS_EXTERNALS_FULLPATH_JSON}/include/nlohmann"
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/irods/common.nix:35]:
```suggestion
    substituteInPlace CMakeLists.txt \
      --replace "DESTINATION usr/bin" "DESTINATION bin" \
      --replace "INCLUDE_DIRS usr/include/" "INCLUDE_DIRS include/" \
      --replace "DESTINATION usr/lib/" "DESTINATION lib/" \
      --replace "{IRODS_EXTERNALS_FULLPATH_JSON}/include" "{IRODS_EXTERNALS_FULLPATH_JSON}/include/nlohmann"
```

File: pkgs/tools/filesystems/irods/default.nix [hunk 2]
```diff
@@ -1,27 +1,29 @@
-{ lib, stdenv, fetchFromGitHub, bzip2, zlib, autoconf, automake, cmake, gnumake, help2man , texinfo, libtool , cppzmq , libarchive, avro-cpp_llvm, boost, jansson, zeromq, openssl , pam, libiodbc, libkrb5, gcc, libcxx, which, catch2 }:
+{ lib, stdenv, fetchFromGitHub, bzip2, zlib, autoconf, automake, cmake, gnumake, help2man , texinfo, libtool , cppzmq , libarchive, avro-cpp_llvm, boost, jansson, zeromq, openssl , pam, libiodbc, libkrb5, gcc, libcxx, which, catch2, nanodbc_llvm, fmt, nlohmann_json, spdlog }:
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/irods/default.nix:1]:
```suggestion
{ lib, stdenv, fetchFromGitHub, bzip2, zlib, autoconf, automake, cmake, gnumake, help2man , texinfo, libtool , cppzmq
, libarchive, avro-cpp_llvm, boost, jansson, zeromq, openssl , pam, libiodbc, libkrb5, gcc, libcxx, which, catch2
, nanodbc_llvm, fmt, nlohmann_json, spdlog
}:
```
The line is way to long or use nixpkgs-fmt to put everything onto its own line.

File: pkgs/tools/filesystems/irods/default.nix [hunk 3]
```diff
@@ -1,27 +1,29 @@
...
 
 let
   avro-cpp=avro-cpp_llvm;
+  nanodbc=nanodbc_llvm;
 in
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/irods/default.nix:6]:
```suggestion
let
  avro-cpp = avro-cpp_llvm;
  nanodbc = nanodbc_llvm;
in
```

```suggestion
```

File: pkgs/tools/filesystems/irods/default.nix [hunk 4]
```diff
@@ -1,27 +1,29 @@
...
 let
   common = import ./common.nix {
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/irods/default.nix:8]: This should use callPackage.

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -6610,7 +6610,9 @@ with pkgs;
   inherit (callPackages ../tools/filesystems/irods rec {
             stdenv = llvmPackages.libcxxStdenv;
             libcxx = llvmPackages.libcxx;
-            boost = boost160.override { inherit stdenv; };
+            boost = boost17x.override { inherit stdenv; };
+            fmt = fmt_8.override { inherit stdenv; };
+            nanodbc_llvm = nanodbc.override { inherit stdenv; };
             avro-cpp_llvm = avro-cpp.override { inherit stdenv boost; };
           })
       irods
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:6618]:
```suggestion
    boost = boost17x.override { inherit stdenv; };
    fmt = fmt_8.override { inherit stdenv; };
    nanodbc_llvm = nanodbc.override { inherit stdenv; };
    avro-cpp_llvm = avro-cpp.override { inherit stdenv boost; };
  })
    irods
```

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -6610,7 +6610,9 @@ with pkgs;
   inherit (callPackages ../tools/filesystems/irods rec {
             stdenv = llvmPackages.libcxxStdenv;
             libcxx = llvmPackages.libcxx;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:6612]:
```suggestion
    stdenv = llvmPackages.libcxxStdenv;
    libcxx = llvmPackages.libcxx;
```

File: pkgs/tools/filesystems/irods/common.nix [hunk 7]
```diff
@@ -28,6 +32,7 @@
     substituteInPlace CMakeLists.txt --replace "DESTINATION usr/bin" "DESTINATION bin"
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/irods/common.nix:32]: The patchShebangs can be combined and this phase should be postPatch.

File: pkgs/tools/filesystems/irods/common.nix [hunk 8]
```diff
@@ -1,10 +1,10 @@
-{ lib, stdenv, bzip2, zlib, autoconf, automake, cmake, gnumake, help2man , texinfo, libtool , cppzmq , libarchive, avro-cpp, boost, jansson, zeromq, openssl, pam, libiodbc, libkrb5, gcc, libcxx, which, catch2 }:
+{ lib, stdenv, bzip2, zlib, autoconf, automake, cmake, gnumake, help2man , texinfo, libtool , cppzmq , libarchive, avro-cpp, boost, jansson, zeromq, openssl, pam, libiodbc, libkrb5, gcc, libcxx, which, catch2, nanodbc, fmt, nlohmann_json, spdlog }:
 
 # Common attributes of irods packages
 
 {
   nativeBuildInputs = [ autoconf automake cmake gnumake help2man texinfo which gcc ];
-  buildInputs = [ bzip2 zlib libtool cppzmq libarchive avro-cpp jansson zeromq openssl pam libiodbc libkrb5 boost libcxx catch2 ];
+  buildInputs = [ bzip2 zlib libtool cppzmq libarchive avro-cpp jansson zeromq openssl pam libiodbc libkrb5 boost libcxx catch2 nanodbc fmt nlohmann_json spdlog ];
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/irods/common.nix:7]:
```suggestion
  buildInputs = [
    bzip2 zlib libtool cppzmq libarchive avro-cpp jansson zeromq openssl pam libiodbc libkrb5 boost
    libcxx catch2 nanodbc fmt nlohmann_json spdlog
  ];
```

File: pkgs/tools/filesystems/irods/common.nix [hunk 9]
```diff
@@ -1,10 +1,10 @@
-{ lib, stdenv, bzip2, zlib, autoconf, automake, cmake, gnumake, help2man , texinfo, libtool , cppzmq , libarchive, avro-cpp, boost, jansson, zeromq, openssl, pam, libiodbc, libkrb5, gcc, libcxx, which, catch2 }:
+{ lib, stdenv, bzip2, zlib, autoconf, automake, cmake, gnumake, help2man , texinfo, libtool , cppzmq , libarchive, avro-cpp, boost, jansson, zeromq, openssl, pam, libiodbc, libkrb5, gcc, libcxx, which, catch2, nanodbc, fmt, nlohmann_json, spdlog }:
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/irods/common.nix:1]:
```suggestion
{ lib, stdenv, bzip2, zlib, autoconf, automake, cmake, gnumake, help2man , texinfo, libtool , cppzmq , libarchive
, avro-cpp, boost, jansson, zeromq, openssl, pam, libiodbc, libkrb5, gcc, libcxx, which, catch2, nanodbc, fmt
, nlohmann_json, spdlog
}:
```

File: pkgs/tools/filesystems/irods/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/tools/filesystems/irods/default.nix:10]: If you do the following you only need to inherit variables which are not defined in top-level.
```suggestion
  common = callPackage ./common.nix {
```

Comment (bzizou) [reply pkgs/tools/filesystems/irods/default.nix]: This fails with something I'm not enough skilled to understand:
```
[bzizou@bart:~/git/nixpkgs]$ nix-build . -A irods-icommands --cores 8 --show-trace
error: while evaluating the attribute 'buildInputs' of the derivation 'irods-icommands-4.2.11' at /home/bzizou/git/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:205:7:
while evaluating the attribute 'override' of the derivation 'irods-4.2.11' at /home/bzizou/git/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:205:7:
cannot coerce a set to a string, at /home/bzizou/git/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:205:7
```

Comment (SuperSandro2000) [reply pkgs/tools/filesystems/irods/default.nix:10]: It would only work if it would be a mkDerivation I think.

**Pull request merged**